### PR TITLE
Adding accounting management feature, feature tests and unit tests

### PR DESCRIPTION
**Pull request by t-kellett**
Pls review

### CODE REVIEW

File: lib/user.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      email: result[0]['email'])
+  end
+
+  attr_reader :id, :name, :email
+
+  def initialize(id:, name:, email:)
+    @id = id
+    @name = name
+    @email = email
+  end
```
Comment (Matt-Warnock) [inline lib/user.rb:36]: These live at the top of the class :-)

File: spec/features/authentication_spec.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
+feature 'authentication' do
```
Comment (Matt-Warnock) [inline spec/features/authentication_spec.rb:1]: Great tests!! very clear to read:-)

File: views/index.erb [hunk 3]
```diff
@@ -0,0 +1,18 @@
+<% if @user %>
```
Comment (Matt-Warnock) [inline views/index.erb:1]: This if else statement could be refactored to just calling
```
<H1><%= @user || 'You can sign up or log in below' # <or just and empty string> %><\h1>
```

File: lib/user.rb (hunk 1 above)
Comment (t-kellett) [reply lib/user.rb:36]: agreed

File: spec/unit/unit_spec.rb [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    end
+  end
+
+  it 'returns nil given an incorrect email address' do
+    expect(User.authenticate(email: 'wrong_email', password: 'password123')).to be_nil
+  end
+
+  it 'returns nil given an incorrect password' do
+    expect(User.authenticate(email: 'test@example.com', password: 'wrongpassword')).to be_nil
+  end
```
[the remark this answers is not in the document]
Comment (t-kellett) [reply spec/unit/unit_spec.rb:34]: Will leave as is as taken directly from the walkthrough

**Pull request closed without merging**